public/pages/CreateTrigger/components/ExpressionQuery/ExpressionQuery.js

```diff
@@ -0,0 +1,306 @@
+    }
+
+    setGraphUi(triggerValues.searchType === 'graph');
+
+    if (selections?.length) {
+      const editorValues = [];
+      selections.map((selection) => {
+        editorValues.push(getQueryTemplate(selection.monitor_id));
+      });
+      const script = editorValues.join(` ${queryConditionOperator} `);
+      setEditorValue(script);
+      _.set(triggerValues, 'triggerDefinitions[0].script.source', script);
+    }
+  }, [value, triggerValues.searchType]);
+
+  const getValue = (expressions) =>
+    expressions.map((exp) => ({
+      condition: _.toLower(exp.description),
+      monitor_id: exp.monitor_id,
+      monitor_name: exp.name,
+    }));
+
+  const changeMonitor = (selection, exp, idx, form) => {
+    const expressions = _.cloneDeep(usedExpressions);
+    expressions[idx] = {
+      ...expressions[idx],
+      monitor_id: selection[0].monitor_id,
+      monitor_name: selection[0].label,
+    };
+
+    setUsedExpressions(expressions);
+    onBlur(form, expressions);
+  };
+
+  const changeCondition = (selection, exp, idx, form) => {
+    const expressions = _.cloneDeep(usedExpressions);
+    expressions[idx] = { ...expressions[idx], description: selection[0].description };
+    setUsedExpressions(expressions);
+    onBlur(form, expressions);
+  };
+
+  const onBlur = (form, expressions) => {
+    form.setFieldTouched('expressionQueries', true);
+    form.setFieldValue(formikName, getValue(expressions));
+    form.setFieldError('expressionQueries', validate());
+  };
+
+  const openPopover = (idx) => {
+    const expressions = _.cloneDeep(usedExpressions);
+    expressions[idx] = { ...expressions[idx], isOpen: !expressions[idx].isOpen };
+    setUsedExpressions(expressions);
+  };
+
+  const closePopover = (idx) => {
+    const expressions = _.cloneDeep(usedExpressions);
+    expressions[idx] = { ...expressions[idx], isOpen: false };
+    setUsedExpressions(expressions);
+  };
+
+  const renderOptions = (expression, idx, form) => (
+    <EuiFlexGroup gutterSize="s" data-test-subj={dataTestSubj}>
+      <EuiFlexItem grow={false}>
+        <EuiComboBox
+          style={{ width: '150px' }}
+          singleSelection={{ asPlainText: true }}
+          compressed
+          selectedOptions={[
+            {
+              label: expression.description,
+              description: expression.description,
+            },
+          ]}
+          onChange={(selection) => changeCondition(selection, expression, idx, form)}
+          options={[
+            { description: '', label: '' },
+            { description: 'AND', label: 'AND' },
+            { description: 'OR', label: 'OR' },
+            { description: 'NOT', label: 'NOT' },
+          ]}
+        />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>{renderMonitorOptions(expression, idx, form)}</EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiButtonIcon
+          data-test-subj={`selection-exp-field-item-remove-${idx}`}
+          onClick={() => {
```

Comment:
define this handler outside and memoize it

Reply:
Done.